### PR TITLE
optimize UserMountCache::registerStorage

### PR DESCRIPTION
**Pull request by icewind1991**
Simplify the logic a bit and use less intermediate arrays

### CODE REVIEW

File: lib/public/Files/Config/ICachedMountInfo.php [hunk 1]
```diff
@@ -84,4 +84,11 @@ public function getRootInternalPath(): string;
 	 * @since 24.0.0
 	 */
 	public function getMountProvider(): string;
+
+	/**
+	 * Get a key that uniquely identifies the mount
+	 *
+	 * @return string
+	 */
+	public function getKey(): string;
```
Comment (blizzz) [inline lib/public/Files/Config/ICachedMountInfo.php:93]: missing @since annotation

**Review by blizzz — CHANGES_REQUESTED** [review]
Missing required annotation, otherwise :+1:

File: lib/public/Files/Config/ICachedMountInfo.php (hunk 1 above)
Comment (Altahrim) [reply lib/public/Files/Config/ICachedMountInfo.php:94]: Just added it :)

**Review by blizzz — APPROVED** [review]

**Review by juliusknorr — APPROVED** [review]

**Review by juliusknorr — CHANGES_REQUESTED** [review]
Ah wait CI shows some issue:

```
PHP Fatal error:  Nesting level too deep - recursive dependency? in /home/runner/work/server/server/lib/private/Files/Config/UserMountCache.php on line 130
```

File: .htaccess [hunk 2]
```diff
@@ -96,3 +96,7 @@
 
 AddDefaultCharset utf-8
 Options -Indexes
+#### DO NOT CHANGE ANYTHING ABOVE THIS LINE ####
```
Comment (juliushaertl) [inline .htaccess:99]: Unrelated :)

Comment (Altahrim) [reply .htaccess:99]: Arf yes, I don't know why it's modified… Thanks!

**Review by juliusknorr — APPROVED** [review]

**Pull request merged**